### PR TITLE
Add search feature

### PR DESCRIPTION
**Pull request by omar2205**
Adds search feature to the home page
image

### CODE REVIEW

**Comment by omar2205** [comment]
The smallest amount of JS to search from recipes list

**Comment by hallzy** [comment]
Short of not having the variables, this is probably the shortest way to do it, and doesn't require continuously removing and adding DOM elements.

```javascript
const rec = document.querySelectorAll('#artlist li')
const search = document.querySelector('#search')

search.addEventListener('change', e => {
    const searchText = e.target.value.toLowerCase();

    rec.forEach(el => {
        const recipeName = el.innerText.toLowerCase();
        const isMatch = recipeName.indexOf(searchText) > -1;

        el.hidden = !isMatch;
    });
});
```

That being said, I will be shocked if this gets into the project.

This search is no different to a standard ctrl+f search, and every issue I have seen created about adding any javascript at all has been closed.

one ref: #150

**Comment by omar2205** [comment]
yes there is the CTRL+F, but to type & search is just easier

**Comment by omar2205** [comment]
image

**Comment by LukeSmithxyz** [comment]
I am open to this. I will test it out when I can.

**Comment by LukeSmithxyz** [comment]
To be clear, yes, search via ctrl-f is possible, but this avoids the issue of visual overwhelm. Users can see all "pasta" entries in one place, rather than ctrl-fing time after time again to different parts of the doc.

Important to me is it maintaining perfect compatibility with cli access.

**Comment by DGKSK8LIFE** [comment]
So you are ok with contributing Javascript to this repo now? @LukeSmithxyz Hmm .

**Review by hallzy — CHANGES_REQUESTED** [review]
If this is something we are going to add, then this is my review. The search does not work for me.

The script is executed before the recipe list is rendered, so the script doesn't know about any recipes.

We will either have:

- Move this into the footer file
- wrap the script in an onload or something, or
- take the script out and put it into a separate `.js` file, and reference it with `<script defer ...>` in this file in the head.

File: templates/article_list_header.html [hunk 1]
```diff
@@ -1,2 +1,31 @@
...
+    const isMatch = recipeName.includes(searchText)
+
+    el.hidden = !isMatch
+  })
+})
+  
+clearSearch.addEventListener('click', e => {
+  search.value = ''
+  rec.forEach(el => el.hidden = false)
+})
```
Comment (hallzy) [inline templates/article_list_header.html:29]:
```suggestion
window.addEventListener('load', () => {
  const rec = document.querySelectorAll('#artlist li');
  const search = document.querySelector('#search');
  const clearSearch = document.querySelector('.clear-search');

  search.addEventListener('change', e => {
    // grab search input value
    const searchText = e.target.value.toLowerCase();
    // for each receipe hide all but matched
    rec.forEach(el => {
      const recipeName = el.innerText.toLowerCase();
      const isMatch = recipeName.includes(searchText);

      el.hidden = !isMatch;
    });
  });

  clearSearch.addEventListener('click', e => {
    search.value = '';
    rec.forEach(el => el.hidden = false);
  });
});
```

This for example, would work.

**Review by hallzy — CHANGES_REQUESTED** [review]

**Review by hallzy — APPROVED** [review]

**Pull request closed without merging**